// spectacle-image-slide/dist/ImageSlide.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = SlideGif;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _spectacle = require('spectacle');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var fullScreenStyle = {
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'column',
  top: 'calc(-50vh + 50%)',
  left: 'calc(-50vw + 50%)',
  position: 'absolute',
  width: '100vw',
  height: '100vh',
  padding: '10vh 10vw'
};

var imageStyle = {
  height: '100%',
  position: 'absolute',
  top: 0,
  left: '50%',
  transform: 'translateX(-50%)'
};

function SlideGif(_ref) {
  var title = _ref.title,
      image = _ref.image,
      titleProps = _ref.titleProps,
      props = _objectWithoutProperties(_ref, ['title', 'image', 'titleProps']);

  return _react2.default.createElement(
    _spectacle.Slide,
    props,
    _react2.default.createElement(
      'div',
      { style: fullScreenStyle },
      !!title && _react2.default.createElement(
        'div',
        { style: { width: 1000 } },
        _react2.default.createElement(
          _spectacle.Heading,
          _extends({
            size: 2,
            caps: true,
            lineHeight: 1,
            textColor: 'secondary',
            style: { marginBottom: '.5em' }
          }, titleProps),
          title
        )
      ),
      _react2.default.createElement(
        'div',
        { style: { flex: 1, position: 'relative' } },
        _react2.default.createElement(_spectacle.Image, {
          style: imageStyle,
          src: image
        })
      )
    )
  );
}

SlideGif.propTypes = {
  image: _propTypes2.default.string,
  title: _propTypes2.default.string,
  titleProps: _propTypes2.default.object
};
