// spectacle-image-slide/dist/index.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory(require('react'), require('prop-types'), require('spectacle')) :
	typeof define === 'function' && define.amd ? define(['react', 'prop-types', 'spectacle'], factory) :
	(global.SpectacleImageSlide = factory(global.React,global.PropTypes,global.Spectacle));
}(this, (function (React,PropTypes,spectacle) { 'use strict';

var React__default = 'default' in React ? React['default'] : React;
PropTypes = 'default' in PropTypes ? PropTypes['default'] : PropTypes;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var fullScreenStyle = {
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'column',
  top: 'calc(-50vh + 50%)',
  left: 'calc(-50vw + 50%)',
  position: 'absolute',
  width: '100vw',
  height: '100vh',
  padding: '10vh 10vw'
};

var imageStyle = {
  height: '100%',
  position: 'absolute',
  top: 0,
  left: '50%',
  transform: 'translateX(-50%)'
};

var ImageSlide$1 = function (_Component) {
  _inherits(ImageSlide, _Component);

  function ImageSlide() {
    _classCallCheck(this, ImageSlide);

    return _possibleConstructorReturn(this, (ImageSlide.__proto__ || Object.getPrototypeOf(ImageSlide)).apply(this, arguments));
  }

  _createClass(ImageSlide, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          title = _props.title,
          image = _props.image,
          titleProps = _props.titleProps,
          props = _objectWithoutProperties(_props, ['title', 'image', 'titleProps']);

      return React__default.createElement(
        'div',
        _extends({ style: fullScreenStyle }, props),
        !!title && React__default.createElement(
          'div',
          { style: { width: 1000 } },
          React__default.createElement(
            spectacle.Heading,
            _extends({
              size: 2,
              caps: true,
              lineHeight: 1,
              textColor: 'secondary',
              style: { marginBottom: '.5em' }
            }, titleProps),
            title
          )
        ),
        React__default.createElement(
          'div',
          { style: { flex: 1, position: 'relative' } },
          React__default.createElement(spectacle.Image, {
            style: imageStyle,
            src: image
          })
        )
      );
    }
  }]);

  return ImageSlide;
}(React.Component);

ImageSlide$1.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  titleProps: PropTypes.object
};

return ImageSlide$1;

})));
